### PR TITLE
[@types/remark-link-card] Add remark-link-card types

### PR DESCRIPTION
**Pull request by miyaji255**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/remark-link-card/remark-link-card-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0a452451577f8f22009ea4027fd4d712b0f0d52d
</p>

File: types/remark-link-card/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+declare namespace rlc {
+    interface Options {
+        /**
+         * Cache image for `next/image` (`bool`, default:`false`)
+         *
+         * Automatically save open graph images and favicon images to `process.cwd()/public/remark-link-card/`.
+         *
+         * Automatically insert the path to images starting with `/remark-link-card/` in the src attribute of img tags.
+         */
+        cache?: boolean;
```
Comment (jakebailey) [inline types/remark-link-card/index.d.ts:10]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Comment (miyaji255) [reply types/remark-link-card/index.d.ts]: Thank you for pointing that out. I missed it. I made the necessary corrections.

File: types/remark-link-card/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        cache?: boolean | undefined;
+
+        /**
+         * Display only hostname of target URL (`bool`, default: `false`)
+         */
+        shortenUrl?: boolean | undefined;
+    }
+}
+
+type rlc = (options?: rlc.Options) => (tree: import("mdast").Root) => Promise<import("mdast").Root>;
```
Comment (jakebailey) [inline types/remark-link-card/index.d.ts:19]: Any reason to not just import the type normally? Like:

```ts
import { Root } from "mdast";

declare const rlc: (options?: rlc.Options) => (tree: Root) => Promise<Root>;
```

File: types/remark-link-card/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        "https://github.com/gladevise/remark-link-card"
+    ],
+    "dependencies": {
+        "@types/mdast": "^4.0.3"
+    },
+    "devDependencies": {
+        "@types/remark-link-card": "workspace:.",
+        "unified": "^11.0.4"
+    },
+    "owners": []
```
Comment (peterblazejewicz) [inline types/remark-link-card/package.json:15]: why empty?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please address the feedback

**Pull request closed without merging**